src/containers/EventCreationPage.tsx

```diff
@@ -82,24 +102,59 @@ const EventCreationPage = () => {
     : {};
 
   const handleComplete = () => {
-    addEvent({
-      variables: {
-        name: eventName,
-        eventDate:
-          dateParts.year &&
-          dateParts.month &&
-          dateParts.day &&
-          `${dateParts.year}-${dateParts.month.padStart(
-            2,
-            '0'
-          )}-${dateParts.day.padStart(2, '0')}`,
-        createdBy: 1,
-        isActive: true,
-      },
-    });
+    if (mode === 'new') {
+      addEvent({
+        variables: {
+          name: eventName,
+          eventDate:
+            dateParts.year &&
+            dateParts.month &&
+            dateParts.day &&
+            `${dateParts.year}-${dateParts.month.padStart(
+              2,
+              '0'
+            )}-${dateParts.day.padStart(2, '0')}`,
+          createdBy: 16, // TODO: change this to proper user
+          isActive: true,
+        },
+      });
+    } else if (mode === 'edit' && eventId) {
+      editEvent({
+        variables: {
+          id: eventId,
+          name: eventName,
+          eventDate:
+            dateParts.year &&
+            dateParts.month &&
+            dateParts.day &&
+            `${dateParts.year}-${dateParts.month.padStart(
+              2,
+              '0'
+            )}-${dateParts.day.padStart(2, '0')}`,
+          createdBy: 16,
+          isActive: true,
+        },
+      });
+    }
     history.replace('/events');
   };
 
+  useEffect(() => {
+    if (!loading && mode === 'edit') {
+      const {
+        name,
+        eventDate,
+      }: {
+        name: string;
+        eventDate: string;
+        id: string;
+      } = data.event;
+
+      setEventName(name);
+      setEventDate(new Date(eventDate));
+    }
+  }, [data]);
```

Comment:
useEffect warning here needs to be addressed

Reply:
good catch I didn't see this, I'll add in `loading` and `mode` to the dependency array (although this shouldn't make a difference much)